### ISSUE TITLE
[3.6 API Refine] post order API improvement proposal

### ISSUE THREAD

**Issue by luzzif** [Loopring/DEX-API#63]
A proposal to present an improved version of the post order API. An adapted version of its OpenAPI (written in YAML) definition can be found [here](https://github.com/Loopring/DEX-API/files/5513166/swaggeroso.txt).

**Comment by yueawang** [comment]
Same suggest, reply status to client is good enough. no need to add order info, accountId, storageId, etc.
BTW: I think order/{stroageId} looks good :), which clearly demonstrates your thinkings on the proposal.

**Comment by luzzif** [comment]
> Same suggest, reply status to client is good enough. no need to add order info, accountId, storageId, etc.
> BTW: I think order/{stroageId} looks good :), which clearly demonstrates your thinkings on the proposal.

Great! Thank you for the feedback. I actually was undecided on whether the API should return the full submitted order or just hash, status and eventually the `idempotent` flag. Guess it makes more sense to go for the second choice, and that generally applies to all `POST` APIs. We should probably avoid returning the fully created entity (which means I'll have to review a few other APIs too).

**Comment by yueawang** [comment]
Yes, Binance has such an option to let users know what happened to their order during the submit, it's less probable for us to add a similar flag, and so far we can reply status only.

**Comment by luzzif** [comment]
I've updated the response. I've left the `hash`, `status`, `idempotent` and the `storageId` in there.
I opted to leave the storage id to make the response totally consistent between the single and multiple orders submission API. In the multiple submission API you'd obviously want to be returned the storage id of every single submitted order in case you want to perform some operations on them.

It's still unclear to me what the `idempotent` attribute in the response is supposed to mean though. Maybe there's a clearer way to express its meaning. I will udate the multiple order submission API accordingly.

**Comment by yueawang** [comment]
> It's still unclear to me what the idempotent attribute in the response is supposed to mean though. Maybe there's a clearer way to express its meaning. I will udate the multiple order submission API accordingly.

idempotent means the same, for example, a user submits the same order twice due to some reasons like network jitter, programming bug, etc, the second will get an OK response with idempotent=true. Maybe we can use an error code to pass this flag, even though the request itself is not an error.
